docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,339 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _command-deduplication:
+
+Command deduplication
+#####################
+
+The interaction of a Daml application with the ledger is inherently asynchronous: applications send commands to the ledger, and some time later they see the effect of that command on the ledger.
+Many things can fail during this time window: the application can crash, the participant node can crash, messages can be lost on the network, or the ledger may be just slow to respond due to a high load.
+If you want to make sure that an intended ledger change is not executed twice, your application needs to robustly handle these failure scenarios.
+This guide covers the following topics:
+
+#. How command deduplication works.
+
+#. How applications can effectively use the command deduplication.
+
+.. _command-dedup-workings:
+
+How command deduplication works
+*******************************
+
+The following fields in a command submissions are relevant for command deduplication.
+The first three form the :ref:`change ID <change-id>` that identifies the intended ledger change.
+
+#. The union of :ref:`party <com.daml.ledger.api.v1.Commands.party>` and :ref:`act_as <com.daml.ledger.api.v1.Commands.act_as>` define the submitting parties.
+  
+#. The :ref:`application ID <com.daml.ledger.api.v1.Commands.application_id>` identifies the application that submits the command.
+
+#. The :ref:`command ID <com.daml.ledger.api.v1.Commands.command_id>` is chosen by the application to identify the intended ledger change.
+
+#. The :ref:`deduplication period <com.daml.ledger.api.v1.Commands.deduplication_period>` specifies the period for which no earlier submissions with the same change ID should have been accepted, as witnessed by a completion event on the :ref:`command completion service <command-completion-service>`.
+   Otherwise, the current submission shall be rejected.
+   The period is specified either as a :ref:`deduplication duration <com.daml.ledger.api.v1.Commands.deduplication_duration>` or as a :ref:`deduplication offset <com.daml.ledger.api.v1.Commands.deduplication_offset>`.
+
+#. The :ref:`submission ID <com.daml.ledger.api.v1.Commands.submission_id>` is returned to the submitting application and allows it to correlate specific submissions to specific completions.
+   An application should never reuse a submission ID.
+
+The ledger may arbitrarily extend the deduplication period specified in the submission, even beyond the maximum deduplication time specified in the :ref:`ledger configuration <ledger-configuration-service>`.
```

Comment:
As explained on in the protobuf, this is the maximum value that the participant/ledger promises to always handle. Since longer deduplication periods are strictly better than shorter ones, the application should be happy if it gets more guarantees.

Reply:
> As explained on in the protobuf, this is the maximum value that the participant/ledger promises to always handle.

I'd also state it here as it's an important guarantee and it would be nice to avoid having to jump to the protobuf, especially considering that this guide is already (and luckily so) very comprehensive.

> longer deduplication periods are strictly better than shorter ones

I wonder if there are use cases where the application would rely on deduplication not to exceed a certain time frame.